mimic/model/nova_objects.py

```diff
@@ -798,6 +798,21 @@ def request_delete(self, http_delete_request, server_id):
         server.update_status(u"DELETED")
         return b''
 
+    def action(self, http_action_request, server_id):
+        """
+        Perform the requested action on the provided server
+        """
+        action_json = loads(http_action_request.content.read())
+        server = self.server_by_id(server_id)
+        if server is None:
+            http_action_request.setResponseCode(500)
+            return b''
+        if 'resize' in action_json:
+            flavor = action_json['resize']['flavorRef']
```

Comment:
If `flavorRef` is not provided, this will fail with `keyerror`. Can we handle that scenario.